File: Core-backend-layer/Programming-fundamentals/Python/OOP/src/inheritance/product.py
class Product:
    def __init__(self, productId, name, price, manufacturer):
        self._productId = productId
        self._name = name
        self._price = price
        self._manufacturer = manufacturer

    # Getter and Setter methods
    def getProductId(self):
        return self._productId

    def getName(self):
        return self._name

    def setName(self, name):
        self._name = name

    def getPrice(self):
        return self._price

    def setPrice(self, price):
        self._price = price

    def getManufacturer(self):
        return self._manufacturer

    def setManufacturer(self, manufacturer):
        self._manufacturer = manufacturer

    def displayInfo(self):
        print(f"Product ID: {self._productId}, Product: {self._name}, Price: ${self._price}, Manufacturer: {self._manufacturer if self._manufacturer else 'N/A'}")

File: Core-backend-layer/Programming-fundamentals/Java/OOP/src/Inheritance/Product.java
public class Product {

    // Fields
    private int productId;
    private String name;
    private double price;
    private String manufacturer;

    public Product(int productId, String name, double price, String manufacturer) {
        this.productId = productId;
        this.name = name;
        this.price = price;
        this.manufacturer = manufacturer;
    }

    // Getter and Setter methods
    public int getProductId() {
        return productId;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    public String getManufacturer() {
        return manufacturer;
    }

    public void setManufacturer(String manufacturer) {
        this.manufacturer = manufacturer;
    }

    public void displayInfo() {
        System.out.println("Product ID: " + productId + ", Product: " + name + ", Price: $" + price
                + ", Manufacturer: " + (manufacturer != null ? manufacturer : "N/A"));
    }
}
